### PR TITLE
Added code for fields with dots(.) in their names work as needed

### LINKED ISSUE #4487: GetFieldValue doesn't work with fields that contain dots(.)
**Describe the bug**

`rnode.GetFieldValue` doesn't work correctly when the target field contains a `.`

For example when I call `rnode.GetFieldValue("foo.bar")` this function will try to find field "bar" within field "foo" instead of finding the field "foo.bar"

**Files that can reproduce the issue**

```go
package v1alpha1

import (
	"testing"

	"github.com/stretchr/testify/require"
	"sigs.k8s.io/kustomize/kyaml/kio"
	"sigs.k8s.io/kustomize/kyaml/yaml"
)

const input = `
kind: Pod
metadata:
  name: hello-world
  labels:
    app: hello-world-app
    foo.appname: hello-world-foo
`

func TestGetFieldValue(t *testing.T) {
	data, err := kio.ParseAll(input)
	require.NoError(t, err)

	labelRNode, err := data[0].Pipe(yaml.Lookup("metadata", "labels"))
	require.NoError(t, err)

	app, err := labelRNode.GetFieldValue("app")
	require.NoError(t, err)
	require.Equal(t, "hello-world-app", app)

	fooAppName, err := labelRNode.GetFieldValue("foo.appname")
	require.NoError(t, err)
	require.Equal(t, "hello-world-foo", fooAppName) // no field named 'foo.appname'
}
```

**Expected output**

`labelRNode.GetFieldValue("foo.appname")` should return `hello-world-app`

**Actual output**

`labelRNode.GetFieldValue("foo.appname")` returns an error because there is no field `foo`

**Kustomize version**

```go.mod
require sigs.k8s.io/kustomize/kyaml v0.13.1
```

**Additional context**

If my pod was defined as:
```yaml
kind: Pod
metadata:
  name: hello-world
  labels:
    app: hello-world-app
    foo.appname: hello-world-foo
    foo:
      appname: wtf       # <--------
```
then it would have "worked" but it would have returned `wtf` instead of `hello-world-app`.

**Workaround**

I can still use `rnode.Pipe(yaml.Lookup("foo.appname"))` and I can also use `m, err := rnode.Map()` followed by `m["foo.appname"]`, it just caught me offguard

### PR DESCRIPTION
**Pull request by annelausf**
Fixes #4487

### CODE REVIEW

**Comment by natasha41575** [comment]
please sign the CLA!

File: api/internal/builtins/AnnotationsTransformer.go [hunk 1]
```diff
@@ -18,7 +16,6 @@ type AnnotationsTransformerPlugin struct {
 	FieldSpecs  []types.FieldSpec `json:"fieldSpecs,omitempty" yaml:"fieldSpecs,omitempty"`
```
Comment (natasha41575) [inline api/internal/builtins/AnnotationsTransformer.go:16]: don't commit this file

File: kyaml/pathsplitterutil/pathsplitter.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2021 The Kubernetes Authors.
 // SPDX-License-Identifier: Apache-2.0
 
-package utils
+package pathsplitterutil
```
Comment (natasha41575) [inline kyaml/pathsplitterutil/pathsplitter.go:4]: let's just call this package "utils", and put it under kyaml/utils

File: kyaml/pathsplitterutil/pathsplitter_test.go [hunk 3]
```diff
@@ -1,13 +1,13 @@
 // Copyright 2021 The Kubernetes Authors.
 // SPDX-License-Identifier: Apache-2.0
 
-package utils_test
+package pathsplitterutil_test
 
 import (
+	. "sigs.k8s.io/kustomize/kyaml/pathsplitterutil"
```
Comment (natasha41575) [inline kyaml/pathsplitterutil/pathsplitter_test.go:7]: Fix the import order, this needs to go down with the github one

File: kyaml/yaml/rnode.go [hunk 4]
```diff
@@ -9,6 +9,7 @@ import (
 	"io/ioutil"
 	"log"
 	"regexp"
+	"sigs.k8s.io/kustomize/kyaml/pathsplitterutil"
```
Comment (natasha41575) [inline kyaml/yaml/rnode.go:12]: same comment here

File: kyaml/yaml/rnode_test.go [hunk 5]
```diff
@@ -2310,7 +2310,7 @@ func TestGetAnnotations(t *testing.T) {
 }
 
 func TestGetFieldValueWithDot(t *testing.T) {
-	t.Skip()
+	//t.Skip()
```
Comment (natasha41575) [inline kyaml/yaml/rnode_test.go:2313]: just remote this line, and the `TODO` below

File: api/filters/replacement/replacement.go [hunk 6]
```diff
@@ -6,6 +6,7 @@ package replacement
 import (
 	"errors"
 	"fmt"
+	utils2 "sigs.k8s.io/kustomize/kyaml/utils"
```
Comment (natasha41575) [inline api/filters/replacement/replacement.go:9]: fix imports, and lets call this something more descriptive, maybe `kyaml_utils` instead of `utils2`

**Comment by natasha41575** [comment]
Please sign the CLA! 

And looks like there are some test failures. If you can narrow down the cause maybe we can help provide guidance to a solution.

**Comment by annelausf** [comment]
Working on signing the CLA and will look into the test case failures in the meantime.

**Comment by annelausf** [comment]
Hi @natasha41575 looked into the errors that we saw and found this. 
I don't see how the changes we made could have caused this. Any ideas?
Screen Shot 2022-04-19 at 2 43 00 PM

**Comment by natasha41575** [comment]
`TestAddManagedbyLabel` might have been caused by some CI changes, I will look into that one.

There are some other tests failing though, that I think are actually caused by this PR, though I don't know why they would be failing. If you have time to investigate the other failures, that would be very helpful.

**Comment by annelausf** [comment]
Waiting to see if the changes we just made will fix the other tests.

**Comment by natasha41575** [comment]
You can see the same test failures in the other PR that attempted to work on this https://github.com/kubernetes-sigs/kustomize/pull/4501, so I think investigation is needed. 

You can run `make verify-kustomize-repo` locally to run the tests.

**Comment by annelausf** [comment]
Also I signed the CLA, however it is still saying Missing CLA Authorization. Is there a way to rerun that check?

**Comment by natasha41575** [comment]
> Also I signed the CLA, however it is still saying Missing CLA Authorization. Is there a way to rerun that check?

/check-cla

If this doesn't work, make sure your email address that you used to sign the CLA is set on your commits.

**Comment by sarjamil** [comment]
just testing

**Comment by sarjamil** [comment]
@sarjamil

**Comment by annelausf** [comment]
@natasha41575 Signed the CLA! Let me know if there are any further comments. Thanks!

**Comment by natasha41575** [comment]
Thank you very much!

/lgtm

**Pull request merged**